spark/src/main/scala/geotrellis/spark/io/avro/codecs/TileCodecs.scala

```diff
@@ -32,7 +35,9 @@ trait TileCodecs {
         .asScala // notice that Avro does not have native support for Short primitive
         .map(_.toShort)
         .toArray
-      new ShortArrayTile(array, rec[Int]("cols"), rec[Int]("rows"))
+      val cellType = CellType.fromString(rec[Utf8]("celltype").toString)
```

Comment:
Actually perhaps even more economical is to include optional NoData value and then inspect it's value to determine if constant or user defined tile is appropriate.